File: player-random.py
#!/usr/bin/python3 -u


import sys
import random
from bid import Bid


instruction = sys.stdin.readline()
while instruction:
    parts = instruction.split(' ')
    if parts[0] == 'NEW_ROUND':
        bid = Bid(0, 6)
    elif parts[0] == 'OPPONENT_BIDS':
        bid = Bid(int(parts[1]), int(parts[2]))
    elif parts[0] == 'PLAY':
        if bid.count > 0 and random.random() < 0.4:
            print('CHALLENGE')
        else:
            print('BID ' + str(bid.count+1) + ' ' + str(bid.value))
    instruction = sys.stdin.readline()

File: PlayerRandom.java
import java.util.Scanner;

public class PlayerRandom {

    public static void main(String[] args) {
        int bidQuantity = 0;
        int bidValue = 6;
        Scanner s = new Scanner(System.in);
        while (s.hasNext()) {
            switch (s.next()) {
                case "NEW_ROUND":
                    bidQuantity = 0;
                    bidValue = 6;
                    break;
                case "PLAYER_BIDS":
                    s.nextInt();
                    bidQuantity = s.nextInt();
                    bidValue = s.nextInt();
                    break;
                case "PLAY":
                    if (bidQuantity > 0 && Math.random() < 0.4) {
                        System.out.println("CHALLENGE");
                    } else {
                        System.out.println("BID " + (bidQuantity + 1) + " " + bidValue);
                    }
            }
            s.nextLine();
        }
    }
}
